### PR TITLE
add change list.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @superyyrrzz @hellosnow @ansyral @qinezh

### CODE REVIEW

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            if (tsvFile == null)
+            {
+                throw new ArgumentNullException(nameof(tsvFile));
+            }
+            if (baseDir == null)
+            {
+                throw new ArgumentNullException(nameof(baseDir));
+            }
+            if (File.Exists(tsvFile))
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs]: >File [](start = 16, length = 4)

add `!`

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+            return ParseCore(tsvFile, baseDir);
+        }
+
+        public void Add(string filePath, ChangeKind kind)
+        {
+            if (filePath == null)
+            {
+                throw new ArgumentNullException(nameof(filePath));
+            }
+            if (filePath.Length == 0)
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs]: >filePath.Length == 0 [](start = 16, length = 20)

merge with above if statement using string.IsNullOrEmpty()?

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            _list.Add(new ChangeItem { FilePath = filePath, Kind = kind });
+        }
+
+        private static ChangeList ParseCore(string tsvFile, string baseDir)
+        {
+            var result = new ChangeList();
+            bool hasError = false;
+
+            foreach (var line in File.ReadLines(tsvFile))
```
Comment (qinezh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs]: Do we need to check null for File.ReadLines(tsvFile)?

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs (hunk 2 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs]: exception is different.

---
In reply to: [73995730](https://github.com/dotnet/docfx/pull/586#discussion_r73995730) [](ancestors = 73995730)

File: src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs (hunk 3 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Build.Engine/Incrementals/ChangeList.cs]: never be null, see: 
[http://referencesource.microsoft.com/#mscorlib/system/io/file.cs](http://referencesource.microsoft.com/#mscorlib/system/io/file.cs)   line 1036
and
[http://referencesource.microsoft.com/#mscorlib/system/io/ReadLinesIterator.cs](http://referencesource.microsoft.com/#mscorlib/system/io/ReadLinesIterator.cs)  line 50

---
In reply to: [73996456](https://github.com/dotnet/docfx/pull/586#discussion_r73996456) [](ancestors = 73996456)

**Comment by qinezh** [comment]
:100:

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vwxyzh** [comment]
#449

**Pull request merged**